test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.jsinterop
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+
+import org.junit.Assert._
+import org.junit.Test
+
+class NonNativeJSTypeTestScala2 {
+  import NonNativeJSTypeTestScala2._
+
+  /** Complements NonNativeJSTypeTest.default_values_for_fields(). */
+  @Test def default_values_for_fields(): Unit = {
+    val obj = new DefaultFieldValues
+
+    /* Value class fields are initialized to null, instead of a boxed
+     * representation of the zero of their underlying types, as for a
+     * Scala class.
+     *
+     * Scala 3 correctly unboxes `null` as the zero of the underlying type of
+     * value classes, which means that this test "fails" in Scala 3, although
+     * it is in fact a progression.
```

Comment:
I find the usage of the term "progression" as an antonym to "regression" quite confusing (it wasn't clear to me what it meant in the commit message). If it is common usage though in dotty / scala land, then we should leave it.

Reply:
Huh. I've seen that often in the scala/scala repo, and thought it was standard terminology. I can use "an improvement" instead, perhaps?

Reply:
"improvement" is definitely clear to me. But if progression is common, then maybe I just needed to learn :)